app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\RegisterRequest;
+use Exception;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Hash;
+use App\Models\User;
+use Illuminate\Support\Facades\Log;
+
+class AuthController extends Controller
+{
+    /**
+     * Register a new user.
+     *
+     * @param RegisterRequest $request The registration request.
+     * @return JsonResponse The JSON response.
+     */
+    public function register(RegisterRequest $request): JsonResponse
+    {
+        $validatedData = $request->validated();
+
+        try {
+            // Create a new user
+            $user = User::create([
```

Comment:
$user = User::create($request->validated());
You may ask, but how then to hash the password before storing it? Here is the answer:
https://gist.github.com/mhanson01/8115466